xarray/backends/h5netcdf_.py

```diff
@@ -465,7 +465,7 @@ def open_datatree(
         use_cftime=None,
         decode_timedelta=None,
         format=None,
-        group: str | Iterable[str] | Callable | None = None,
+        group: str | Callable[[], str] | None = None,
```

Comment:
Is that really the correct Callable signature? Does Callable actually work at all?

Reply:
I don't think so. However, this is something I will discuss in the next datatree meeting.

Reply:
`Callable` was a suggestion I had some time ago, but it also is unsupported right now. So I'd also remove that, and possibly add it back when/if we add support for `Iterable[str]`.